lnrpc/walletrpc/walletkit_server.go

```diff
@@ -822,16 +825,9 @@ func (w *WalletKit) ListSweeps(ctx context.Context,
 		sweepTxDetails = append(sweepTxDetails, tx)
 	}
 
-	// Fail if we have not retrieved all of our sweep transactions from the
-	// wallet.
-	if len(sweepTxDetails) != len(txids) {
-		return nil, fmt.Errorf("not all sweeps found by list "+
-			"transactions: %v, %v", len(sweepTxDetails), len(txids))
-	}
-
 	return &ListSweepsResponse{
 		Sweeps: &ListSweepsResponse_TransactionDetails{
-			TransactionDetails: lnrpc.RPCTransactionDetails(transactions),
+			TransactionDetails: lnrpc.RPCTransactionDetails(sweepTxDetails),
```

Comment:
I think bot verbose and non-verbose response should be filtered, such that they are consistent (returning the same sweeps but with different level of detail). The linked bug seems to recommend the same. WDYT?

Reply:
Originally the point of having the non-verbose output was to save us the call to list all of our transactions from the wallet, which was my thinking in still returning the full list (incl rbf). 

I also think there can be a use-case for knowing the txids of sweeps that never made it to chain (eg, the case where you have alerting on every tx your node broadcasts, you'd want to be able to check lnd's sweeps to whitelist the tx).

But I do agree it's confusing to have different output for verbose/not. Could add a `IncludeDropped` option which just returns all the sweeps when verbose is false, and does not work if verbose is true so that you can still get those rbf-ed txids if you need them? Unsure about it tbh.

Reply:
IMHO simplifying and only including sweeps that made it to the chain would make more sense. Having the `IncludeDropped` smells like something users probably ain't going to need (although a good generalization idea).

Reply:
Yeah SGTM, keep it consistent for now and then add the invisible/rbfed sweeps if anybody requires them. Will make a note in the proto that these sweeps aren't listed.